api/api.py

```diff
@@ -139,10 +139,135 @@ async def node_traceroutes(request: Request, id: str) -> JSONResponse:
         async def chat(request: Request) -> JSONResponse:
             return jsonable_encoder(self.data.chat)
 
+        @app.get("/v1/telemetry")
+        async def telemetry(request: Request) -> JSONResponse:
+            return jsonable_encoder(self.data.telemetry[:1000])
+        
+        @app.get("/v1/traceroutes")
+        async def traceroutes(request: Request) -> JSONResponse:
+            return jsonable_encoder(self.data.traceroutes[:1000])
+
+        @app.get("/v1/messages")
+        async def messages(request: Request) -> JSONResponse:
+            return jsonable_encoder(self.data.messages[:1000])
+        
+        @app.get("/v1/mqtt_messages")
+        async def mqtt_messages(request: Request) -> JSONResponse:
+            return jsonable_encoder(self.data.mqtt_messages[:1000])
+        
+        @app.get("/v1/stats")
+        async def stats(request: Request) -> JSONResponse:
+            stats = {
+                'active_nodes': 0,
+                'total_chat': len(self.data.chat['channels']['0']['messages']),
+                'total_nodes': len(self.data.nodes),
+                'total_messages': len(self.data.messages),
+                'total_mqtt_messages': len(self.data.mqtt_messages),
+                'total_telemetry': len(self.data.telemetry),
+                'total_traceroutes': len(self.data.traceroutes),
+            }
+            for _, node in self.data.nodes.items():
+                if 'active' in node and node['active']:
+                    stats['active_nodes'] += 1
+        
+            return jsonable_encoder({"stats": stats})        
+
         @app.get("/v1/server/config")
         async def server_config(request: Request) -> JSONResponse:
-            # TODO: Sanitize config (i.e. username, password, api_key, etc)
-            return jsonable_encoder({ "config": self.config })
+            # # TODO: Sanitize config (i.e. username, password, api_key, etc)
+            # sanitized_config = self.config
+            # del sanitized_config['broker']['host']
+            # del sanitized_config['broker']['port']
+            # del sanitized_config['broker']['username']
+            # del sanitized_config['broker']['password']
+            
+            # del sanitized_config['channels']['encryption_key']
+
+            # del sanitized_config['paths']
+
+            # sanitized_config['integrations']['discord'] = {
+            #     'enabled': self.config['integrations']['discord']['enabled'],
+            # }
+            # sanitized_config['integrations']['geocoding'] = {
+            #     'enabled': self.config['integrations']['geocoding']['enabled'],
+            #     'provider': self.config['integrations']['geocoding']['provider'],
+            # }
+            
+            whitelist = {
```

Comment:
My thought process behind the whitelist is that we want to be inclusionary, not exclusionary. For instance, if I added another api key, I know I'm going to forget to add it to a blacklist, whereas if it's a whitelist, we'll be able to more easily ensure that we're not accidentally leaking data.

Reply:
Except that we're adding more non-secured config values than secured ones, and it will be easy for people to forget this. With secured values, it in general requires more thought and so adding to the list will be more obvious.

I just don't like the idea of having to mimic a full structure that will be changing a lot.

Reply:
That's fair, i guess i was erring on the side of security + we don't really make a ton of use of the config endpoint other than to fetch the server's node id. I'm good switching this to a blacklist if you really want me to.